src/pattern-matching/let-control-flow/let-else.md

```diff
@@ -7,19 +7,19 @@ off the end of the block).
 
 ```rust,editable
 fn hex_or_die_trying(maybe_string: Option<String>) -> Result<u32, String> {
-    // TODO: The structure of this code is difficult to follow -- rewrite it with let-else!
-    if let Some(s) = maybe_string {
-        if let Some(first_byte_char) = s.chars().next() {
-            if let Some(digit) = first_byte_char.to_digit(16) {
-                Ok(digit)
-            } else {
-                Err(String::from("not a hex digit"))
-            }
-        } else {
-            Err(String::from("got empty string"))
-        }
-    } else {
-        Err(String::from("got None"))
+    let s = match maybe_string {
+        Some(s) => s,
+        None => return Err(String::from("got None")),
+    };
```

Comment:
What if we strike a balance here:
```suggestion
    let s = if let Some(s) = maybe_string {
        s
    } else {
        return Err(String::from("got None")),
    };
```

Reply:
I'm not sure how this helps? Using `if let` spreads it across more lines vertically, but doesn't demonstrate any more nesting. I'd say using `match` here is clearer, but I guess I don't feel that strongly about it.

Reply:
Fundamentally, the thing that people get into here is "I need to pattern-match some enums, with an early exit if they don't match". I think the `let .. = if .. else { return .. };` is a clearer representation of this desire than the match, since match doesn't represent the "else" as clearly.

Reply:
> since match doesn't represent the "else" as clearly.

... isn't the `None` case the same as the `else` case? Are you saying that the `None` case in a `match` might be less clear to students than `else` is, since they'd be more familiar with `if else` coming from other languages?

Reply:
Yes, I'm saying both of those things :)

Reply:
Okay, I've changed it to use `if let` instead. @fw-immunant does this seem any better to you? The starting code still doesn't have any nesting, but we're now using `if let` instead of `match`, which maybe helps address your point about the motivation for `let else`?